### PR TITLE
Implement Local Secondary Index Support

### LINKED ISSUE #20: Local Secondary Indexes
It doesn't look like Local Secondary Indexes are supported.  Is that just coincidental, or was it attempted and deemed too difficult?

Since a LSI must be part of table creation, if you need one, you can't use this module (as opposed to GSI's, which can be added after table creation.)

If it's just coincidental, I may attempt to implement it and PR it back in.

### PR DESCRIPTION
**Pull request by jamisonhyatt**
resolves #20

### CODE REVIEW

**Comment by aknysh** [comment]
@jamisonhyatt  thanks a lot!
we'll review this.
was it tested on AWS?
also, please run `terraform fmt` (out CI is failing)

**Comment by jamisonhyatt** [comment]
> was it tested on AWS?

yes - tested with no LSI, an LSI, and an LSI + GSI.

> also, please run `terraform fmt` (out CI is failing)

done.

File: examples/complete/main.tf [hunk 1]
```diff
@@ -22,6 +22,25 @@ module "dynamodb_table" {
...
     },
+    {
+      name = "Timestamp"
+      type = "S"
+    },
+  ]
+
+  local_secondary_index_map = [
+    {
+      name               = "TimstampSortIndex"
```
Comment (aknysh) [inline examples/complete/main.tf]: fix typo `TimestampSortIndex`

File: main.tf [hunk 2]
```diff
@@ -37,6 +37,15 @@ resource "null_resource" "global_secondary_indexe_names" {
   triggers = "${map("name", lookup(var.global_secondary_index_map[count.index], "name"))}"
 }
 
+resource "null_resource" "local_secondary_indexe_names" {
```
Comment (aknysh) [inline main.tf]: fix typo `local_secondary_index_names`

Comment (jamisonhyatt) [reply main.tf]: Done (FWIW - it's mispelled on the GSI null resource above)

File: examples/complete/main.tf (hunk 1 above)
Comment (jamisonhyatt) [reply examples/complete/main.tf]: done

File: main.tf (hunk 2 above)
Comment (aknysh) [reply main.tf]: can you please fix that as well?

Comment (jamisonhyatt) [reply main.tf]: done

**Pull request merged**